src/mscorlib/src/System/Span.cs

```diff
@@ -164,5 +167,145 @@ public bool Equals(Span<T> other)
             return (_length == other._length) &&
                 (_length == 0 || JitHelpers.ByRefEquals(ref JitHelpers.GetByRef<T>(ref _rawPointer), ref JitHelpers.GetByRef<T>(ref other._rawPointer)));
         }
+
+        /// <summary>
+        /// Copies the contents of this span into destination span. The destination
+        /// must be at least as big as the source, and may be bigger.
+        /// </summary>
+        /// <param name="destination">The span to copy items into.</param>
+        public bool TryCopyTo(Span<T> destination)
+        {
+            if (Length > destination.Length)
+                return false;
+
+            SpanHelper.CopyTo<T>(ref destination._rawPointer, ref _rawPointer, Length);
+            return true;
+        }
+
+        /// <summary>
+        /// Copies the contents of this span into destination array. The destination
+        /// must be at least as big as the source, and may be bigger.
+        /// </summary>
+        /// <param name="destination">The array to copy items into.</param>
+        public bool TryCopyTo(T[] destination)
+        {
+            return TryCopyTo(new Span<T>(destination));
+        }
+
+        /// <summary>
+        /// Copies the contents of this span into destination memory. 
+        /// Only Value Types that contain no pointers are supported.
+        /// </summary>
+        /// <param name="destination">An unmanaged pointer to memory.</param>
+        /// <param name="elementsCount">The number of T elements the memory contains.</param>
+        [CLSCompliant(false)]
+        public unsafe bool TryCopyTo(void* destination, int elementsCount)
+        {
+            if (Length > (uint)elementsCount || JitHelpers.ContainsReferences<T>())
+            {
+                return false;
+            }
+
+            SpanHelper.Memmove<T>((byte*)destination, (byte*)_rawPointer, elementsCount);
+            return true;
+        }
+
+        public void Set(ReadOnlySpan<T> values)
+        {
+            SpanContracts.RequiresInInclusiveRange(values._length, _length);
+
+            SpanHelper.CopyTo<T>(ref _rawPointer, ref values._rawPointer, values.Length);
+        }
+
+        public void Set(T[] values)
+        {
+            Set(new ReadOnlySpan<T>(values));
+        }
+
+        [CLSCompliant(false)]
+        public unsafe void Set(void* source, int elementsCount)
+        {
+            SpanContracts.RequresNoReferences<T>();
+            SpanContracts.RequiresNonNegative(elementsCount);
+            SpanContracts.RequiresInInclusiveRange(_length, elementsCount);
+
+            SpanHelper.Memmove<T>((byte*)_rawPointer, (byte*)source, elementsCount);
+        }
+    }
+
+    internal static class SpanContracts
+    {
+        internal static void RequiresIndexInRange(int index, int length) {
+            if ((uint)index >= (uint)length)
```

Comment:
I do not think that having contract methods on the side is the preferred corefx coding style. We have done experiments with code contracts in the past, so some old code may still look similar to this ... but it was abandoned as not a very good idea.